niworkflows/interfaces/ants.py

```diff
@@ -0,0 +1,283 @@
+    'ThresholdImage input.nii.gz output.nii.gz 0.50000 1.00000 1.00000 0.00000'
+
+    >>> res = ThresholdImage(dimension=3)
+    >>> res.inputs.input_image = 'input.nii.gz'
+    >>> res.inputs.output_image = 'output.nii.gz'
+    >>> res.inputs.mode = 'Kmeans'
+    >>> res.inputs.num_thresholds = 4
+    'ThresholdImage input.nii.gz output.nii.gz Kmeans 4'
+
+    """
+    _cmd = 'ThresholdImage'
+    input_spec = ThresholdImageInputSpec
+    output_spec = ThresholdImageOutputSpec
+
+
+class AIInputSpec(ANTSCommandInputSpec):
+    dimension = traits.Int(3, usedefault=True, argstr='-d %d',
+                           desc='dimension of output image')
+    verbose = traits.Bool(False, usedefault=True, argstr='-v %d',
+                          desc='enable verbosity')
+
+    fixed_image = traits.File(
+        exists=True, mandatory=True,
+        desc='Image to which the moving_image should be transformed')
+    moving_image = traits.File(
+        exists=True, mandatory=True,
+        desc='Image that will be transformed to fixed_image')
+
+    fixed_image_mask = traits.File(
+        exists=True, argstr='-x %s', desc='fixed mage mask')
+    moving_image_mask = traits.File(
+        exists=True, requires=['fixed_image_mask'],
+        desc='moving mage mask')
+
+    metric_trait = (
+        traits.Enum("Mattes", "GC", "MI"),
+        traits.Int(32),
+        traits.Enum('Regular', 'Random', 'None'),
+        traits.Range(value=0.2, low=0.0, high=1.0)
+    )
+    metric = traits.Tuple(*metric_trait, argstr='-m %s', mandatory=True,
+                          desc='the metric(s) to use.')
+
+    transform = traits.Tuple(
+        traits.Enum('Affine', 'Rigid', 'Similarity'),
+        traits.Range(value=0.1, low=0.0, exclude_low=True),
+        argstr='-t %s[%f]', usedefault=True,
+        desc='Several transform options are available')
+
+    principal_axes = traits.Bool(False, usedefault=True, argstr='-p %d', xor=['blobs'],
+                                 desc='align using principal axes')
+    search_factor = traits.Tuple(
+        traits.Float(20), traits.Range(value=0.12, low=0.0, high=1.0),
+        usedefault=True, argstr='-s [%f,%f]', desc='search factor')
+
+    search_grid = traits.Either(
+        traits.Tuple(traits.Float, traits.Tuple(traits.Float, traits.Float, traits.Float)),
+        traits.Tuple(traits.Float, traits.Tuple(traits.Float, traits.Float)),
+        argstr='-g %s', desc='Translation search grid in mm')
+
+    convergence = traits.Tuple(
+        traits.Range(low=1, high=10000, value=10),
+        traits.Float(1e-6),
+        traits.Range(low=1, high=100, value=10),
+        usedefault=True, argstr='-c [%d,%f,%d]', desc='convergence')
+
+    output_transform = traits.File(
+        'initialization.mat', usedefault=True, argstr='-o %s',
+        desc='output file name')
+
+
+class AIOuputSpec(base.TraitedSpec):
+    output_transform = traits.File(exists=True, desc='output file name')
+
+
+class AI(ANTSCommand):
```

Comment:
Is this an initial affine registration? Didn't we already have something like `AffineInitializer`?

In any event, it should be clearer what `antsAI` does.

Reply:
It is a replacement for AffineInitializer as of ANTs 2.1.0.